src/spxmod/regmod_builder.py

```diff
@@ -289,27 +290,44 @@ def objective(self, coefs: NDArray) -> float:
         weights = self.data.weights * self.data.trim_weights
         y = self.get_lin_param(coefs)
 
+        z = np.exp(-y)
+        z_inf = np.isinf(z)
+        if z_inf.sum() > 0:
+            log_term = np.empty_like(z)
+            log_term[z_inf] = -y[z_inf]
+            log_term[~z_inf] = np.log(1 + z[~z_inf])
+        else:
+            log_term = np.log(1 + z)
+
         prior_obj = self.objective_from_gprior(coefs)
-        likli_obj = weights.dot(
-            np.log(1 + np.exp(-y)) + (1 - self.data.obs) * y
-        )
-        return prior_obj + likli_obj
+        likli_obj = weights.dot(log_term + (1 - self.data.obs) * y)
+        fun = prior_obj + likli_obj
+
+        if np.isinf(fun):
+            raise ValueError("objective blew up")
+        return fun
 
     def gradient(self, coefs: NDArray) -> NDArray:
         mat = self.mat[0]
         weights = self.data.weights * self.data.trim_weights
-        z = np.exp(self.get_lin_param(coefs))
+        y = self.get_lin_params(coefs)
 
         prior_grad = self.gradient_from_gprior(coefs)
-        likli_grad = mat.T.dot(weights * (z / (1 + z) - self.data.obs))
+        likli_grad = mat.T.dot(weights * (expit(y) - self.data.obs))
+
         return prior_grad + likli_grad
 
     def hessian(self, coefs: NDArray) -> Matrix:
         mat = self.mat[0]
         weights = self.data.weights * self.data.trim_weights
-        z = np.exp(self.get_lin_param(coefs))
-        likli_hess_scale = weights * (z / ((1 + z) ** 2))
+        y = self.get_lin_params(coefs)
+
+        z = np.empty_like(y)
+        y_pos = y > 0
+        z[y_pos] = np.exp(-y[y_pos])
+        z[~y_pos] = np.exp(y[~y_pos])
 
+        likli_hess_scale = weights * (z / ((1 + z) ** 2))
```

Comment:
Similarly here, we might be able to use `expit` function twice.
```python
import numpy as np
from scipy.special import expit

y = ...
assert np.allclose(np.exp(-y) / (1 + np.exp(-y)) ** 2, expit(y) * expit(-y))
```

Reply:
Great!